debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9VTablesCommand.java

```diff
@@ -52,45 +55,58 @@ public void run(String command, String[] args, Context context, final PrintStrea
+				startIndex = 0;
+			} else {
+				vTable = J9ClassHelper.oldVTable(ramClass);
+				vTableMethodCount = vTable.at(0).add(1);
+				jitVTableOffset = vTableMethodCount.longValue() * UDATA.SIZEOF;
+				startIndex = 2;
+			}
 
 			if(J9BuildFlags.interp_nativeSupport) {
-			    if (!vm.jitConfig().isNull()) {
-			        jitVTable = UDATAPointer.cast(U8Pointer.cast(ramClass).sub((vTableSlotCount.longValue()+1) * UDATA.SIZEOF));
-//			        jitVTableSlotCount = vTableSlotCount.sub(1);
-			    }
+				if (!vm.jitConfig().isNull()) {
+					jitVTable = UDATAPointer.cast(U8Pointer.cast(ramClass).sub(jitVTableOffset));
+				}
 			}
 	
-			CommandUtils.dbgPrint(out, String.format("VTable for j9class %s  (size=%d - 1 for skipped resolve method)\n", ramClass.getHexAddress(), vTableSlotCount.longValue()));
+			CommandUtils.dbgPrint(out, String.format("VTable for j9class %s  (size=%d - 1 for skipped resolve method)\n", ramClass.getHexAddress(), vTableMethodCount.longValue()));
 			CommandUtils.dbgPrint(out, String.format("\tInterpreted%s\n", (!jitVTable.isNull()) ? "\t\tJitted\n" : ""));
 			
-			/*	First entry in vtable is the vtable count.
-			 * 	Second entry is the magic method reference.
-			 *  Skip both and start from the third element in vTable which is the first virtual method reference.
-			 **/
-		    for (long i = 2; i < vTableSlotCount.longValue() + 1; i++) {
-		    	String name = J9MethodHelper.getName(J9MethodPointer.cast(vTable.at(i)));		    	
-		    	String intAddr = U8Pointer.cast(vTable.at(i)).getHexAddress();
-		        if (!jitVTable.isNull()) {		            
-		        	String jitAddr = U8Pointer.cast(jitVTable.at(vTableSlotCount.sub(i))).getHexAddress();
-		        	CommandUtils.dbgPrint(out, String.format(" %d\t!j9method %s\t%s\t%s\n", i-1, intAddr, jitAddr, name));
-		        } else {
-		        	CommandUtils.dbgPrint(out, String.format(" %d\t!j9method %s\t%s\t\n", i-1, intAddr, name));
-		        }
-		    }
-		    return;
+			/* Start from the first element after vTable header which is the first virtual method reference.
+			 * 
+			 * For Old VTable:
+			 * 		First entry in vtable is the vtable count.
+			 * 		Second entry is the magic method reference.
+			 * 		Skip both and start from the third element in vTable which is the first virtual method reference.
+			 */
+			for (long i = startIndex; i < vTableMethodCount.longValue(); i++) {
+				String name = J9MethodHelper.getName(J9MethodPointer.cast(vTable.at(i)));		    	
+				String intAddr = U8Pointer.cast(vTable.at(i)).getHexAddress();
+				if (!jitVTable.isNull()) {		            
+					String jitAddr = U8Pointer.cast(jitVTable.at(vTableMethodCount.sub(i))).getHexAddress();
+					CommandUtils.dbgPrint(out, String.format(" %d\t!j9method %s\t%s\t%s\n", i, intAddr, jitAddr, name));
+				} else {
+					CommandUtils.dbgPrint(out, String.format(" %d\t!j9method %s\t%s\t\n", i, intAddr, name));
```

Comment:
this change will make ddr vtable command `<!j9vtables>` with old core files to start display at index 2, and new cores start at index 0 (currently starts from 1)